packages/core/src/shared/lsp/lspResolver.ts

```diff
@@ -202,7 +217,7 @@ export class LanguageServerResolver {
 
         const fetchTasks = contents.map(async (content) => {
             return {
-                res: await new HttpResourceFetcher(content.url, { showUrl: true }).get(),
+                res: await new HttpResourceFetcher(content.url, { showUrl: true, timeout: timeout }).get(),
```

Comment:
This is the expected pattern 👍 

Alternatively, if `HttpResourceFetcher.cancel()` exists, that could be called from the `onCancellationRequested` handler (line  93).